resources/api.go

```diff
@@ -95,21 +98,85 @@ type Api interface {
 }
 
 type api struct {
-	name   string
-	routes map[string]Route
-	m      *manager
+	name          string
+	routes        map[string]Route
+	m             *manager
+	securityRules map[string]interface{}
+	security      map[string][]string
 }
 
-func (m *manager) NewApi(name string) Api {
-	return &api{
+func (m *manager) NewApi(name string, opts ...ApiOption) (Api, error) {
+	rsc, err := m.resourceServiceClient()
+
+	if err != nil {
```

Comment:
can we cuddle the error checks please?

```
err := foo()
if err != nil {
    return err
}
```
This is used in all the golang repos I have seen.

Reply:
fixed